app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.browser
+
+import android.content.Context
+import android.support.design.widget.AppBarLayout
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.View
+
+import org.mozilla.focus.R
+import org.mozilla.focus.widget.AnimatedProgressBar
+
+/**
+ * The toolbar of the BrowserFragment; displaying the URL and other controls.
+ */
+class DisplayToolbar : AppBarLayout, AppBarLayout.OnOffsetChangedListener {
+    private lateinit var toolbarContent: View
+    private lateinit var progressView: AnimatedProgressBar
+
+    private val collapsedProgressTranslationY: Float by lazy {
+        TypedValue.applyDimension(
+                TypedValue.COMPLEX_UNIT_DIP, 2f, resources.displayMetrics)
+    }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        toolbarContent = findViewById<View>(R.id.toolbar_content)
```

Comment:
nit: As I mentioned in the previous commit, with kotlin android extensions, these lines are no longer necessary

Reply:
I'll look at the extensions in #1411 and will update this part in a follow-up patch.